### DISCUSSION TITLE
CLI not compatible

### DISCUSSION THREAD

**Discussion by iam-steve** [twilio-professional-services/flex-project-template#68] [Q&A] [answered]
After cloning the current version of the template and following the install steps... when we try and run the local instance of the plugin (twilio flex:plugins:start), we're getting the error "The plugin plugin-flex-ts-template version (v5) is not compatible with this CLI command." 

Any one have insights as to what might be wrong?

**Comment by jared-hunter** [comment, marked as answer]
which plugin are you using, flex v1 or flex v2?

flex v2's package.json recently updated 

devDependencies
    "twilio-cli": "3.4.2", -> "twilio-cli":"4.2.0"
    
there might be some incompatibility with whats installed globally to whats installed locally

what versions of the cli are you seeing when you execute `twilio version`

**Reply by iam-steve** [reply]
Thanks for the quick response!

We're using Flex v2... and we did a fresh clone of the repo today to run some tests, so it should be using the updated package.json 

When I run `twilio version` it get `twilio-cli/5.1.0 darwin-x64 node-v16.17.1`

**Reply by jared-hunter** [reply]
hi steve, np!

it looks like there's maybe some issues going on with package versions installed local, i just ran into a similar thing when reviewing the latest PR which updates flex plugin scripts to v6.0.2 except it was"The plugin plugin-flex-ts-template version (v6) is not compatible with this CLI command."

to get you going id recommend doing the following

- take a clean checkout of the repository (a new feature was just applied with some package dependency updates as mentioned above)
- [update your twilio cli ](https://www.twilio.com/docs/twilio-cli/getting-started/update) confirming that you are on v5.2.0 with `twilio version`
- update your twilio flex plugins script to 6.2.0 - you can confirm which version you are running if you run `twilio plugins`.  There are multiple ways to update this and it depends how you installed it in the first place.  I tried...
    - `twilio plugins:update` but it didnt actually update it
    - `twilio flex:plugins:upgrade-plugin --install` - this did not work either
    - `twilio plugins:install @twilio-labs/plugin-flex@6.0.2` - this worked for me and `twilio plugins` finally reported the correct version


let me know if this helped

**Reply by iam-steve** [reply]
Will give this a try over the weekend and let you know how it goes. Thanks again!

**Reply by iam-steve** [reply]
So, it looks like I was able to update the twilio CLI and confirmed with "twilio version"  `twilio-cli/5.2.0 darwin-x64 node-v14.15.1` -- however, I am now getting errors when I try and update the plugin to 6.0.2 it is saying `The engine "node" is incompatible with this module. Expected version ">=14.16". Got "14.15.1" error Found incompatible module.`

**Reply by iam-steve** [reply]
- hmmm, okay, so I updated the node version and was able to successfully install plugin-flex@6.0.2... 
- The problem now is when I run `twilio flex:plugins:start` I get the following error `» twilio-cli encountered an unexpected error. To report this issue, execute the command with the "-l debug" flag, then copy the output to a new issue here: "https://github.com/twilio/flex-plugin-builder/issues"`

**Reply by iam-steve** [reply]
whoops -- my mistake with the above reply -- forgot to `npm install` again. :) 

Think I'm making progress now.
